spring-pulsar/src/test/java/org/springframework/pulsar/core/DefaultSchemaResolverTests.java

```diff
@@ -305,13 +310,15 @@ void primitiveKeyValueMessageType() {
 			}
 
 			@Test
-			void customKeyValueMessageTypeDefaultsToBytesSchema() {
-				ResolvableType kvType = ResolvableType.forClassWithGenerics(KeyValue.class, Foo.class, Bar.class);
+			void customKeyValueMessageTypeDefaultsToJSONSchema() {
+				ResolvableType kvType = ResolvableType.forClassWithGenerics(KeyValue.class, Foo.class, Zaa.class);
 				assertThat(resolver.getSchema(SchemaType.NONE, kvType))
 						.asInstanceOf(InstanceOfAssertFactories.type(KeyValueSchema.class))
 						.satisfies((keyValueSchema -> {
-							assertThat(keyValueSchema.getKeySchema()).isEqualTo(Schema.BYTES);
-							assertThat(keyValueSchema.getValueSchema()).isEqualTo(Schema.BYTES);
+							assertThat(keyValueSchema.getKeySchema().getSchemaInfo())
```

Comment:
So this change in defaults REALLY helps out in the key-value case as this prevents having to specify the key-value and type-value mappings.